driver-core/src/test/unit/com/mongodb/Jep395RecordCodecProviderTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledForJreRange;
+import org.junit.jupiter.api.condition.JRE;
+
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class Jep395RecordCodecProviderTest {
+
+    @Test
+    @EnabledForJreRange(min = JRE.JAVA_17)
+    void canSupportJavaRecordsOnJava17() {
+        assertTrue(new Jep395RecordCodecProvider().hasRecordSupport());
```

Comment:
Testing: 

```
        Jep395RecordCodecProvider jep395RecordCodecProvider = new Jep395RecordCodecProvider();
        try {
            Class<?> recordClass = Class.forName("java.lang.Record");
            assertNotNull(jep395RecordCodecProvider.get(recordClass, MongoClientSettings.getDefaultCodecRegistry()));
        } catch (ClassNotFoundException e) {
            fail("Should not fail on JAVA 17");
        }
```

Failed because using `recordClass`always returned null  - it failed the clazz.isRecord in the actual `RecordCodecProvider`.

Reply:
Oh, interesting.